### PR TITLE
[DEM-1198] Check tag type in mongo and memory storage

### PR DESCRIPTION
**Pull request by qSaevar**

### CODE REVIEW

File: src/qilib/utils/storage/interface.py [hunk 1]
```diff
@@ -109,6 +109,13 @@ def _tag_to_string(tag: Union[str, List[str]]) -> str:
             tag = '/'.join(tag)
         return tag
 
+    @staticmethod
+    def _validate_tag(tag: List[str]) -> None:
```
Comment (peendebak) [inline src/qilib/utils/storage/interface.py]: If you want to be fancy
```suggestion
   TagType = List[str]
    def _validate_tag(tag: TagType ) -> None:
```
and use TagType everywhere...

File: src/qilib/utils/storage/interface.py [hunk 2]
```diff
@@ -109,6 +109,13 @@ def _tag_to_string(tag: Union[str, List[str]]) -> str:
...
+        """ Assert that tag is a list of strings."""
+
+        if not isinstance(tag, list) or not all(type(item) == str for item in tag):
```
Comment (peendebak) [inline src/qilib/utils/storage/interface.py]: Use `isinstance(item, str)` instead?

File: src/tests/unittests/utils/storage/test_storage_memory.py [hunk 3]
```diff
@@ -83,6 +83,10 @@ def test_save_load(self):
         results = storage.list_data_subtags(['1', '2'])
         self.assertEqual(results, ['3'])
 
+    def test_save_mixed_list_races_error(self):
```
Comment (peendebak) [inline src/tests/unittests/utils/storage/test_storage_memory.py]:
```suggestion
    def test_save_tag_mixed_list_raises_error(self):
```

File: src/tests/unittests/utils/storage/test_storage_mongo.py [hunk 4]
```diff
@@ -116,6 +116,10 @@ def test_save_load(self):
     def test_save_with_wrong_tag_type(self):
         self.assertRaises(TypeError, self.storage.save_data, None, 'wrong/tag/type')
 
+    def test_save_mixed_list_races_error(self):
```
Comment (peendebak) [inline src/tests/unittests/utils/storage/test_storage_mongo.py]:
```suggestion
    def test_save_tag_mixed_list_raises_error(self):
```

File: src/qilib/utils/storage/interface.py [hunk 5]
```diff
@@ -109,6 +109,13 @@ def _tag_to_string(tag: Union[str, List[str]]) -> str:
...
+            raise TypeError('Tag should be a list of strings')
```
Comment (peendebak) [inline src/qilib/utils/storage/interface.py]:
```suggestion
            raise TypeError(f'Tag {tag} should be a list of strings')
```

**Comment by qSaevar** [comment]
@peendebak The branch was out of date and has now been updated, can you re-approve it?

**Pull request merged**